qbraid/runtime/native/result.py

```diff
@@ -88,3 +88,16 @@ def backend_version(self) -> str:
     def seed(self) -> Optional[int]:
         """Returns the seed used for the simulation."""
         return self._seed
+
+
+class NECVectorAnnealerResultData(AnnealingResultData):
+    """Class for storing and accessing the results of an annealing job.
+    Args:
+        ResultData (ResultData): ABC for runtime results linked to a specific ExperimentType.
+
+    """
+
+    def __init__(self, sa_results: Optional[list[dict[str, Any]]] = None):
+        """Create a new NECVectorAnnealerResultData instance."""
+        super().__init__(sa_results=sa_results)
+        self._sa_results = sa_results or []
```

Comment:
redundant `self._sa_results = sa_results or []` if passed to super class with same code don't need it here. But confused will this `sa_result` list data type be universal to all `AnnealingResultData`? Or is a data type specific to NEC? If not universal, should not be present in parent class.